### ISSUE TITLE
I am trying to create a conditional deformable atals with conditional segmentation map. Is there any way to create the conditoinal segmentation map for the conditonal atlas?

### ISSUE THREAD

**Issue by bingcheg** [voxelmorph/voxelmorph#470]
**Task** (what are you trying to do/register?)

[please describe task here]

**What have you tried**

Please describe specifics of your approach // use of `vxm`

**Details of experiments**

Please carefully specify details about your experiments. If you are training, when what is the setup? What loss are you using? What does the convergence look like? If you are registering, please show example inputs and outputs. etc.

**Comment by adalca** [comment]
This is an active reseach project for us led by @mariannerakic. You would need to modify the existing code quite a bit, but basically you should replace (or add) the images to segmentation maps and change the loss functions to be segmentation-based. See the semi-supervised voxelmorph implementation for that, it should give you a good starting point.

**Comment by bingcheg** [comment]
> This is an active reseach project for us led by @mariannerakic. You would need to modify the existing code quite a bit, but basically you should replace (or add) the images to segmentation maps and change the loss functions to be segmentation-based. See the semi-supervised voxelmorph implementation for that, it should give you a good starting point.

Thanks for replying me. I am confused that after you creating the conditional template based on age, for example, how do you examine the quality of your created conditional template? after I read the paper, I think it is based on dice score, but it requres the segmentation map, so that is why I want to know how to get the segmentation map and how to quantitively examine the created conditional atlas. 

Many thanks for helping me with this.

**Comment by bingcheg** [comment]
H, I read your paper about 'Learning Conditional Deformable Templates
with Convolutional Networks'. I am bit confused about this session part:

![image](https://user-images.githubusercontent.com/69607047/188057254-82ab75f1-fd96-4d33-81d9-1992040ceed3.png)

**Comment by bingcheg** [comment]
how do you "averaging their warped segementations"?

**Comment by adalca** [comment]
if you have 10 segmented subjects, you do for each:
- register each subject to the atlas --> this gives the def field phi
- propagate the label map (preferably in one-hot format) to the atlas using phi --> this gives you warped seg map

Then take the average over the 10 one-hot warped segmentation maps (you could weight these by the age difference between the subject and the atlas.

**Comment by bingcheg** [comment]
> if you have 10 segmented subjects, you do for each:
> 
> * register each subject to the atlas --> this gives the def field phi
> * propagate the label map (preferably in one-hot format) to the atlas using phi --> this gives you warped seg map
> 
> Then take the average over the 10 one-hot warped segmentation maps (you could weight these by the age difference between the subject and the atlas.

Hi thanks for replying me, I just wonder how to take the average? using ants tool or just simply using python code to average it?

**Comment by adalca** [comment]
python code, np.mean and such
